client/app/bundles/course/admin/pages/AnnouncementsSettings/operations.ts

```diff
@@ -0,0 +1,35 @@
+import { AxiosError } from 'axios';
+
+import CourseAPI from 'api/course';
+import {
+  AnnouncementsSettingsData,
+  AnnouncementsSettingsPostData,
+} from 'types/course/admin/announcements';
+
+type Data = Promise<AnnouncementsSettingsData>;
+type ProbableData = Promise<AnnouncementsSettingsData | undefined>;
+
+export const fetchAnnouncementsSettings = async (): Data => {
+  const response = await CourseAPI.admin.announcements.index();
+  return response.data;
+};
+
+export const updateAnnouncementsSettings = async (
+  data: AnnouncementsSettingsData,
+): ProbableData => {
+  const adaptedData: AnnouncementsSettingsPostData = {
+    settings_announcements_component: {
+      title: data.title,
+    },
+  };
+
+  try {
```

Comment:
As discussed offline, I extracted the error response data in `operations.ts` because I did not want to let our views 'know' that we are using Axios; kind of adhering to the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter).

I will, however, revise and handle the throwing better.